backend/benefit/applications/tests/test_applications_api.py

```diff
@@ -1216,21 +1220,105 @@ def test_attachment_delete(request, api_client, application, status, expected_co
 
 
 @pytest.mark.parametrize(
-    "status",
+    "status,upload_result",
     [
-        ApplicationStatus.DRAFT,
-        ApplicationStatus.ADDITIONAL_INFORMATION_NEEDED,
+        (ApplicationStatus.DRAFT, 201),
+        (ApplicationStatus.ADDITIONAL_INFORMATION_NEEDED, 201),
+        (ApplicationStatus.HANDLING, 403),
+        (ApplicationStatus.ACCEPTED, 403),
+        (ApplicationStatus.REJECTED, 403),
     ],
 )
-def test_pdf_attachment_upload(request, api_client, application, status):
+def test_pdf_attachment_upload_and_download_as_applicant(
+    request, api_client, application, status, upload_result
+):
     application.status = status
     application.save()
     response = _upload_pdf(request, api_client, application)
-    assert response.status_code == 201
+    assert response.status_code == upload_result
+    if upload_result != 201:
+        return
     assert len(application.attachments.all()) == 1
     attachment = application.attachments.all().first()
     assert attachment.attachment_type == AttachmentType.EMPLOYMENT_CONTRACT
     assert attachment.attachment_file.name.endswith(".pdf")
+    response = api_client.get(get_detail_url(application))
+    assert len(response.data["attachments"]) == 1
+    assert response.data["attachments"][0]["attachment_file"].startswith(
+        "http://testserver/"
+    )
+    assert (
+        "handlerapplications" not in response.data["attachments"][0]["attachment_file"]
+    )
+    # url_path = urlparse(response.data['attachments'][0]['attachment_file']).path
+    file_dl = api_client.get(response.data["attachments"][0]["attachment_file"])
+    assert file_dl.status_code == 200
+    bytes = file_dl.getvalue()
+    assert bytes[:4].decode("utf-8") == "%PDF"
```

Comment:
Are you checking the contents of the file directly? Hmm... if you want to know whether the file is a PDF file or not then perhaps there are better options for that?

Reply:
My bad, I googled about PDF file format and Adobe's [PDF Reference](https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/pdf_reference_archives/PDFReference.pdf#page=83) File Header section says:

> **3.4.1 File Header**
> 
> The first line of a PDF file is a *header* identifying the version of the PDF specification to which the file conforms. For a file conforming to PDF version 1.4, the header should be
> 
> &nbsp;&nbsp;&nbsp;&nbsp;%PDF-1.4
> 
> However, since any file conforming to an earlier version of PDF also conforms to version 1.4, an application that processed PDF 1.4 can also accept files with any of the following headers:
> 
> &nbsp;&nbsp;&nbsp;&nbsp;%PDF-1.0
> &nbsp;&nbsp;&nbsp;&nbsp;%PDF-1.1
> &nbsp;&nbsp;&nbsp;&nbsp;%PDF-1.2
> &nbsp;&nbsp;&nbsp;&nbsp;%PDF-1.3

So checking for the file starting with "%PDF" is a valid although quite rudimentary way to check whether the file is a PDF file.